docs/adrs/0003-bulk-import-ui.md

```diff
@@ -0,0 +1,98 @@
+> 1. Alter the app specs to support multiple in-line parameters while retaining backward compatibility (including documentation, etc.).
+> 2. Use a distinct app spec for importers that can include these options.
+> 3. Hard-code how all import apps look and feel in the Narrative interface itself.
+>
+> If we decide to update how app specs work, that means working in the `narrative_method_store` and `catalog` repos as well as internal documentation.
+
+## Alternatives Considered
+
+* Implement the UI in the existing App Cell codebase
+* Implement the UI in a new Cell type
+* Don't use a cell, but some other interface
+
+## Decision Outcome
+
+## Consequences
+
+## Pros and Cons of the Alternatives
+
+### Implement the UI in the existing App Cell code base
+
+This option would entail adding a different mode for the current App Cell, likely based on some additional metadata feature.
+
+* `+` Reduce spread of additional, yet similar, cell types with similar code.
+* `+` Reuse existing cell control mechanisms currently in place with minimal modification.
+* `+` Reuse existing wrapping UI elements - cell title, icons, tabs, etc.
+* `-` Work done to the App cell needs to be made backward compatible with existing functionality.
+* `-` Any "breaking" work (i.e. refactor, etc.) will need to be complete before any releases are possible, making intermediate releases possibly challenging.
+* `-` Adding yet another mode for the App cell will increase the maintenance burden on an already complex code base - see especially the FSM.
+* `-` The draft UI is very complex and may turn the existing code into a deep series of nested logic.
+
+### Implement the UI in a new Cell type
+
+This option will create a new nbextension to serve a new cell type as an extension to the code cell.
+
+* `+` This would mean working with a fresh codebase without existing constraints.
+* `+` Intermediate, minimally functional changes, can be rapidly prototyped and released.
+* `-` To stick with the DRY principle, this would mean refactoring parts of the current app cell to make various components importable (message bus responses, job state management, tab menu, etc.) instead of their integration (really we might want to consider that anyway).
+* `-` Much of the structure of the mockup already functions similar to the existing App Cell and might become duplicated work with additional maintenance burden.
+* `-` This would mean updating other components that make use of Narratives - the Static Narrative interface, Search, and the Dashboard, for example, all introspect Narratives and handle cells based on their type.
```

Comment:
```
This would mean updating other components that make use of Narratives - the Static Narrative interface, Search, and the Dashboard, for example, all introspect Narratives and handle cells based on their type.
```
Wouldn't we have to do this anyway with both approaches?

Reply:
Probably. But the existing approach just reports as an app cell - a new approach might break things or require an update to search to get the new cells indexed properly.